File: typescript/src/creational/factory/Factory.ts
import { Notification } from "./interface/IBase"; 
import { Email } from "./notifications/Email";
import { Push } from "./notifications/Push";
import { SMS } from "./notifications/Sms";

export function createNotification(type: string): Notification{
    type = type.toLowerCase().trim().replace(/\s+/g, "");

    if (type == "email"){
        return new Email()
    }
    else if(type == "sms"){
        return new SMS()
    }
    else if(type == "push"){
        return new Push()
    }
    else{
        throw new Error("Notification type not supported.")
    }

} 

File: python/creational/factory/factory.py
from notifications.base import Notification
from notifications.email import Email
from notifications.sms import SMS
from notifications.push import Push


def create_notification(type: str) -> Notification:
    type = type.lower().strip().replace(" ", "")

    if type == "email":
        return Email()
    elif type == "sms":
        return SMS()
    elif type == "push":
        return Push()
    else:
        raise ValueError("Notification type not supported")